views/pages/ViewChartPage/index.jsx

```diff
@@ -116,6 +135,13 @@ const ViewChartPage = () => {
       setGraph(newGraph)
     })
   }, [chart_id, search])
+  const visibleRows = React.useMemo(
+    () =>
+      graph?.graphTable?.tableRows
+        .slice(page * rowsPerPage, page * rowsPerPage + rowsPerPage)
+        .sort(comparator),
+    [sortDirection, sortBy, page, rowsPerPage, graph?.graphTable]
+  )
```

Comment:
```suggestion
  const visibleRows = () => {
    if (!graph) return

    return graph.graphTable.tableRows
      .slice(page * rowsPerPage, page * rowsPerPage + rowsPerPage)
      .sort(comparator)
  }
```

Reply:
I prefer to memoize it and track these dependencies

Reply:
my point here is that you can short-circuit and return an empty array if there is no `graph`. Then you can remove the safe navigators. My point about the memoization is in the comment thread above.

Reply:
oh okay, so with this commit we don't have to track the "complex objects" any more and we don't use safe navigators. 
[c014801](https://github.com/AMP-SCZ/dpdash/pull/695/commits/c0148010f3c405c14fa944ad28c4484d85267e02)

Reply:
https://github.com/AMP-SCZ/dpdash/pull/695#discussion_r1535788146

Reply:
[94b11e1](https://github.com/AMP-SCZ/dpdash/pull/695/commits/94b11e110f46375b8941d93b40bfcd12a6b2b624)